### ISSUE TITLE
Error when opening child-collections on v0.38.0-rc2

### ISSUE THREAD

**Issue by LukeAbell** [metabase/metabase#14444]
**Describe the bug**
After upgrading to v0.38.0-rc2, child-collections cannot be opened.

**Logs**

```
app-main.bundle.js?765f61c164024acbbaec:5 Uncaught TypeError: Cannot read property 'length' of undefined
    at app-main.bundle.js?765f61c164024acbbaec:5
    at t.value (app-main.bundle.js?765f61c164024acbbaec:5)
    at c._renderValidatedComponentWithoutOwnerOrContext (vendor.bundle.js?765f61c164024acbbaec:485)
    at c._renderValidatedComponent (vendor.bundle.js?765f61c164024acbbaec:485)
    at c.performInitialMount (vendor.bundle.js?765f61c164024acbbaec:485)
    at c.mountComponent (vendor.bundle.js?765f61c164024acbbaec:485)
    at Object.mountComponent (vendor.bundle.js?765f61c164024acbbaec:92)
    at c.performInitialMount (vendor.bundle.js?765f61c164024acbbaec:485)
    at c.mountComponent (vendor.bundle.js?765f61c164024acbbaec:485)
    at Object.mountComponent (vendor.bundle.js?765f61c164024acbbaec:92)
```

**To Reproduce**
1. Open parent collection
2. Open child collection

**Expected behavior**
Child collection collections display

**Screenshots**
<img width="1676" alt="screenshot_243517" src="https://user-images.githubusercontent.com/3343530/104957392-152d6980-599c-11eb-8197-442575455965.png">

**Information about your Metabase Installation:**
```json
{
  "browser-info": {
    "language": "en-US",
    "platform": "MacIntel",
    "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 11_1_0) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/87.0.4280.141 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.7+10",
    "java.vendor": "AdoptOpenJDK",
    "java.vendor.url": "https://adoptopenjdk.net/",
    "java.version": "11.0.7",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.7+10",
    "os.name": "Linux",
    "os.version": "4.19.150+",
    "user.language": "en",
    "user.timezone": "UTC"
  },
  "metabase-info": {
    "databases": [
      "sqlserver",
      "postgres",
      "mysql",
      "googleanalytics"
    ],
    "hosting-env": "unknown",
    "application-database": "postgres",
    "application-database-details": {
      "database": {
        "name": "PostgreSQL",
        "version": "9.6.18"
      },
      "jdbc-driver": {
        "name": "PostgreSQL JDBC Driver",
        "version": "42.2.8"
      }
    },
    "run-mode": "prod",
    "version": {
      "date": "2021-01-15",
      "tag": "v0.38.0-rc2",
      "branch": "master",
      "hash": "72d0e88"
    },
    "settings": {
      "report-timezone": "UTC"
    }
  }
}
```

**Severity**
Blocking upgrade to v38

**Comment by flamber** [comment]
Probably related to #14280
![0 37 7](https://user-images.githubusercontent.com/1447303/105360351-c92b3080-5bf8-11eb-81da-5ce670e3e0de.png)
![0 38 0rc2 - and - master](https://user-images.githubusercontent.com/1447303/105360367-cc262100-5bf8-11eb-91e6-c5ccad80129b.png)

**Comment by flamber** [comment]
From what I can see, this is caused by the order of the collections in the application database.
When I move "Parent collection" to somewhere else (ie. "DB1"), then all the sub-collections show up correctly. Same when I move the collection back to root.
Looking through `collection` and `permissions*` tables, the only difference between not-working and working is the order of the collections.
```diff
--- COLLECTION_202101211714.sql	2021-01-21 17:31:50.826765995 +0100
+++ COLLECTION_202101211718.sql	2021-01-21 17:31:51.850781260 +0100
@@ -1,30 +1,30 @@
 INSERT INTO PUBLIC.COLLECTION (ID,NAME,DESCRIPTION,COLOR,ARCHIVED,LOCATION,PERSONAL_OWNER_ID,SLUG,NAMESPACE) VALUES (1,'Meta Base''s Personal Collection',NULL,'#31698A',false,'/',1,'meta_base_s_personal_collection',NULL);
 INSERT INTO PUBLIC.COLLECTION (ID,NAME,DESCRIPTION,COLOR,ARCHIVED,LOCATION,PERSONAL_OWNER_ID,SLUG,NAMESPACE) VALUES (3,'Test2 Test2''s Personal Collection',NULL,'#31698A',false,'/',2,'test2_test2_s_personal_collection',NULL);
 INSERT INTO PUBLIC.COLLECTION (ID,NAME,DESCRIPTION,COLOR,ARCHIVED,LOCATION,PERSONAL_OWNER_ID,SLUG,NAMESPACE) VALUES (4,'DB1',NULL,'#509EE3',false,'/',NULL,'db1',NULL);
 INSERT INTO PUBLIC.COLLECTION (ID,NAME,DESCRIPTION,COLOR,ARCHIVED,LOCATION,PERSONAL_OWNER_ID,SLUG,NAMESPACE) VALUES (11,'Sub-PC',NULL,'#509EE3',false,'/1/',NULL,'sub_pc',NULL);
 INSERT INTO PUBLIC.COLLECTION (ID,NAME,DESCRIPTION,COLOR,ARCHIVED,LOCATION,PERSONAL_OWNER_ID,SLUG,NAMESPACE) VALUES (21,'Usage Test Collection 10',NULL,'#509EE3',false,'/4/',NULL,'usage_test_collection_10',NULL);
 INSERT INTO PUBLIC.COLLECTION (ID,NAME,DESCRIPTION,COLOR,ARCHIVED,LOCATION,PERSONAL_OWNER_ID,SLUG,NAMESPACE) VALUES (22,'Usage Test Collection 11',NULL,'#509EE3',false,'/4/',NULL,'usage_test_collection_11',NULL);
 INSERT INTO PUBLIC.COLLECTION (ID,NAME,DESCRIPTION,COLOR,ARCHIVED,LOCATION,PERSONAL_OWNER_ID,SLUG,NAMESPACE) VALUES (23,'Usage Test Collection 12',NULL,'#509EE3',false,'/4/',NULL,'usage_test_collection_12',NULL);
 INSERT INTO PUBLIC.COLLECTION (ID,NAME,DESCRIPTION,COLOR,ARCHIVED,LOCATION,PERSONAL_OWNER_ID,SLUG,NAMESPACE) VALUES (24,'Usage Test Collection 13',NULL,'#509EE3',false,'/4/',NULL,'usage_test_collection_13',NULL);
 INSERT INTO PUBLIC.COLLECTION (ID,NAME,DESCRIPTION,COLOR,ARCHIVED,LOCATION,PERSONAL_OWNER_ID,SLUG,NAMESPACE) VALUES (9,'Usage Test Collection 02',NULL,'#509EE3',false,'/4/',NULL,'usage_test_collection_02',NULL);
 INSERT INTO PUBLIC.COLLECTION (ID,NAME,DESCRIPTION,COLOR,ARCHIVED,LOCATION,PERSONAL_OWNER_ID,SLUG,NAMESPACE) VALUES (10,'Usage Test Collection 03',NULL,'#509EE3',false,'/4/',NULL,'usage_test_collection_03',NULL);
 INSERT INTO PUBLIC.COLLECTION (ID,NAME,DESCRIPTION,COLOR,ARCHIVED,LOCATION,PERSONAL_OWNER_ID,SLUG,NAMESPACE) VALUES (15,'Usage Test Collection 04',NULL,'#509EE3',false,'/4/',NULL,'usage_test_collection_04',NULL);
 INSERT INTO PUBLIC.COLLECTION (ID,NAME,DESCRIPTION,COLOR,ARCHIVED,LOCATION,PERSONAL_OWNER_ID,SLUG,NAMESPACE) VALUES (16,'Usage Test Collection 05',NULL,'#509EE3',false,'/4/',NULL,'usage_test_collection_05',NULL);
 INSERT INTO PUBLIC.COLLECTION (ID,NAME,DESCRIPTION,COLOR,ARCHIVED,LOCATION,PERSONAL_OWNER_ID,SLUG,NAMESPACE) VALUES (17,'Usage Test Collection 06',NULL,'#509EE3',false,'/4/',NULL,'usage_test_collection_06',NULL);
 INSERT INTO PUBLIC.COLLECTION (ID,NAME,DESCRIPTION,COLOR,ARCHIVED,LOCATION,PERSONAL_OWNER_ID,SLUG,NAMESPACE) VALUES (18,'Usage Test Collection 07',NULL,'#509EE3',false,'/4/',NULL,'usage_test_collection_07',NULL);
 INSERT INTO PUBLIC.COLLECTION (ID,NAME,DESCRIPTION,COLOR,ARCHIVED,LOCATION,PERSONAL_OWNER_ID,SLUG,NAMESPACE) VALUES (19,'Usage Test Collection 08',NULL,'#509EE3',false,'/4/',NULL,'usage_test_collection_08',NULL);
 INSERT INTO PUBLIC.COLLECTION (ID,NAME,DESCRIPTION,COLOR,ARCHIVED,LOCATION,PERSONAL_OWNER_ID,SLUG,NAMESPACE) VALUES (20,'Usage Test Collection 09',NULL,'#509EE3',false,'/4/',NULL,'usage_test_collection_09',NULL);
 INSERT INTO PUBLIC.COLLECTION (ID,NAME,DESCRIPTION,COLOR,ARCHIVED,LOCATION,PERSONAL_OWNER_ID,SLUG,NAMESPACE) VALUES (25,'Usage Test Collection 14',NULL,'#509EE3',false,'/4/',NULL,'usage_test_collection_14',NULL);
 INSERT INTO PUBLIC.COLLECTION (ID,NAME,DESCRIPTION,COLOR,ARCHIVED,LOCATION,PERSONAL_OWNER_ID,SLUG,NAMESPACE) VALUES (26,'X Slow embedding',NULL,'#509EE3',false,'/',NULL,'x_slow_embedding',NULL);
 INSERT INTO PUBLIC.COLLECTION (ID,NAME,DESCRIPTION,COLOR,ARCHIVED,LOCATION,PERSONAL_OWNER_ID,SLUG,NAMESPACE) VALUES (27,'Test3 Test3-renamed2''s Personal Collection',NULL,'#31698A',false,'/',6,'test3_test3_renamed2_s_personal_collection',NULL);
 INSERT INTO PUBLIC.COLLECTION (ID,NAME,DESCRIPTION,COLOR,ARCHIVED,LOCATION,PERSONAL_OWNER_ID,SLUG,NAMESPACE) VALUES (8,'Usage Test Collection 01 alpha / omega',NULL,'#509EE3',false,'/4/',NULL,'usage_test_collection_01_alpha___omega',NULL);
-INSERT INTO PUBLIC.COLLECTION (ID,NAME,DESCRIPTION,COLOR,ARCHIVED,LOCATION,PERSONAL_OWNER_ID,SLUG,NAMESPACE) VALUES (28,'Child collection',NULL,'#509EE3',false,'/5/',NULL,'child_collection',NULL);
-INSERT INTO PUBLIC.COLLECTION (ID,NAME,DESCRIPTION,COLOR,ARCHIVED,LOCATION,PERSONAL_OWNER_ID,SLUG,NAMESPACE) VALUES (29,'Another child',NULL,'#509EE3',false,'/5/',NULL,'another_child',NULL);
-INSERT INTO PUBLIC.COLLECTION (ID,NAME,DESCRIPTION,COLOR,ARCHIVED,LOCATION,PERSONAL_OWNER_ID,SLUG,NAMESPACE) VALUES (30,'Sub-child',NULL,'#509EE3',false,'/5/28/',NULL,'sub_child',NULL);
-INSERT INTO PUBLIC.COLLECTION (ID,NAME,DESCRIPTION,COLOR,ARCHIVED,LOCATION,PERSONAL_OWNER_ID,SLUG,NAMESPACE) VALUES (31,'Another sub-child',NULL,'#509EE3',false,'/5/29/',NULL,'another_sub_child',NULL);
-INSERT INTO PUBLIC.COLLECTION (ID,NAME,DESCRIPTION,COLOR,ARCHIVED,LOCATION,PERSONAL_OWNER_ID,SLUG,NAMESPACE) VALUES (32,'Sub-child2',NULL,'#509EE3',false,'/5/28/',NULL,'sub_child2',NULL);
-INSERT INTO PUBLIC.COLLECTION (ID,NAME,DESCRIPTION,COLOR,ARCHIVED,LOCATION,PERSONAL_OWNER_ID,SLUG,NAMESPACE) VALUES (5,'Parent collection',NULL,'#509EE3',false,'/',NULL,'parent_collection',NULL);
 INSERT INTO PUBLIC.COLLECTION (ID,NAME,DESCRIPTION,COLOR,ARCHIVED,LOCATION,PERSONAL_OWNER_ID,SLUG,NAMESPACE) VALUES (33,'New collection',NULL,'#509EE3',false,'/',NULL,'new_collection',NULL);
 INSERT INTO PUBLIC.COLLECTION (ID,NAME,DESCRIPTION,COLOR,ARCHIVED,LOCATION,PERSONAL_OWNER_ID,SLUG,NAMESPACE) VALUES (34,'New sub collection',NULL,'#509EE3',false,'/33/',NULL,'new_sub_collection',NULL);
 INSERT INTO PUBLIC.COLLECTION (ID,NAME,DESCRIPTION,COLOR,ARCHIVED,LOCATION,PERSONAL_OWNER_ID,SLUG,NAMESPACE) VALUES (2,'Test Collection',NULL,'#509EE3',false,'/',NULL,'test_collection',NULL);
 INSERT INTO PUBLIC.COLLECTION (ID,NAME,DESCRIPTION,COLOR,ARCHIVED,LOCATION,PERSONAL_OWNER_ID,SLUG,NAMESPACE) VALUES (12,'Sub-test collection (contains all the questions) / with slash',NULL,'#509EE3',false,'/2/',NULL,'sub_test_collection__contains_all_the_questions____with_slash',NULL);
+INSERT INTO PUBLIC.COLLECTION (ID,NAME,DESCRIPTION,COLOR,ARCHIVED,LOCATION,PERSONAL_OWNER_ID,SLUG,NAMESPACE) VALUES (5,'Parent collection',NULL,'#509EE3',false,'/',NULL,'parent_collection',NULL);
+INSERT INTO PUBLIC.COLLECTION (ID,NAME,DESCRIPTION,COLOR,ARCHIVED,LOCATION,PERSONAL_OWNER_ID,SLUG,NAMESPACE) VALUES (28,'Child collection',NULL,'#509EE3',false,'/5/',NULL,'child_collection',NULL);
+INSERT INTO PUBLIC.COLLECTION (ID,NAME,DESCRIPTION,COLOR,ARCHIVED,LOCATION,PERSONAL_OWNER_ID,SLUG,NAMESPACE) VALUES (29,'Another child',NULL,'#509EE3',false,'/5/',NULL,'another_child',NULL);
+INSERT INTO PUBLIC.COLLECTION (ID,NAME,DESCRIPTION,COLOR,ARCHIVED,LOCATION,PERSONAL_OWNER_ID,SLUG,NAMESPACE) VALUES (30,'Sub-child',NULL,'#509EE3',false,'/5/28/',NULL,'sub_child',NULL);
+INSERT INTO PUBLIC.COLLECTION (ID,NAME,DESCRIPTION,COLOR,ARCHIVED,LOCATION,PERSONAL_OWNER_ID,SLUG,NAMESPACE) VALUES (32,'Sub-child2',NULL,'#509EE3',false,'/5/28/',NULL,'sub_child2',NULL);
+INSERT INTO PUBLIC.COLLECTION (ID,NAME,DESCRIPTION,COLOR,ARCHIVED,LOCATION,PERSONAL_OWNER_ID,SLUG,NAMESPACE) VALUES (31,'Another sub-child',NULL,'#509EE3',false,'/5/29/',NULL,'another_sub_child',NULL);
```

**Comment by camsaul** [comment]
Fixed by #14280
